narwhal/worker/src/batch_maker.rs

```diff
@@ -211,5 +268,60 @@ impl BatchMaker {
             .created_batch_latency
             .with_label_values(&[self.committee.epoch.to_string().as_str(), reason])
             .observe(self.batch_start_timestamp.elapsed().as_secs_f64());
+
+        // Clone things to not capture self
+        let store = self.store.clone();
+        let worker_id = self.id;
+        let tx_digest = self.tx_digest.clone();
+        let metadata = batch.metadata.clone();
+
+        async move {
```

Comment:
nit: I would look into extracting this to a method so we can clearly state the intention of the future and reduce the size of the `seal` method.

Reply:
I am scared that making this a method would capture a &self, making it difficult to then have a &mut in the main select loop. Lets do this later if at all.